src/main/io/serial_4way.c

```diff
@@ -464,36 +463,34 @@ void esc4wayProcess(serialPort_t *mspPort)
         // restart looking for new sequence from host
         do {
             CRC_in.word = 0;
-            timedOut = ReadByteCrc(&ESC, ESC_TIMEOUT_US);
-        } while ((ESC != cmd_Local_Escape) && !timedOut);
+            // No timeout as BLHeliSuite32 has this loops sitting indefinitely waiting for input
+            ReadByteCrc(&ESC, 0);
+        } while (ESC != cmd_Local_Escape);
 
         RX_LED_ON;
 
         Dummy.word = 0;
         O_PARAM = &Dummy.bytes[0];
         O_PARAM_LEN = 1;
 
+        timedOut = ReadByteCrc(&CMD, CMD_TIMEOUT_US) ||
+                   ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US) ||
+                   ReadByteCrc(&ioMem.D_FLASH_ADDR_L, ARG_TIMEOUT_US) ||
+                   ReadByteCrc(&I_PARAM_LEN, ARG_TIMEOUT_US);
+
         if (!timedOut) {
-            timedOut = ReadByteCrc(&CMD, CMD_TIMEOUT_US);
-            timedOut |= ReadByteCrc(&ioMem.D_FLASH_ADDR_H, ARG_TIMEOUT_US);
-            timedOut |= ReadByteCrc(&ioMem.D_FLASH_ADDR_L, ARG_TIMEOUT_US);
-            timedOut |= ReadByteCrc(&I_PARAM_LEN, ARG_TIMEOUT_US);
-
-            if (!timedOut) {
-                InBuff = ParamBuf;
-                uint8_t i = I_PARAM_LEN;
-                do {
-                  timedOut |= ReadByteCrc(InBuff, DAT_TIMEOUT_US);
-                  InBuff++;
-                  i--;
-                } while ((i > 0) && !timedOut);
-
-                if (!timedOut) {
-                    timedOut |= ReadByte(&CRC_check.bytes[1], CRC_TIMEOUT_US);
-                    timedOut |= ReadByte(&CRC_check.bytes[0], CRC_TIMEOUT_US);
-                }
+            uint8_t i = I_PARAM_LEN;
+
+            InBuff = ParamBuf;
+            do {
+              timedOut = ReadByteCrc(InBuff++, DAT_TIMEOUT_US);
+            } while ((--i > 0) && !timedOut);
+
+            for (int8_t i = 1; (i >= 0) && !timedOut; i--) {
```

Comment:
Just asking - was this supposed to be `int8_t i = 0` ???

Reply:
No, it's counting down; 1 then 0.